### PR TITLE
Add fi_cq_err_entry::src_addr

### PR DESCRIPTION
**Pull request by iziemba**
fi_cq_err_entry::src_addr can be used by CQ errors to report a source address. How this field will be used to CQ error event specific.

### CODE REVIEW

**Review by shefty — APPROVED** [review]
Need approval from the provider maintainers, but looks good.

**Comment by shijin-aws** [comment]
bot:aws:retest

File: prov/psm3/src/psmx3_cq.c [hunk 1]
```diff
@@ -954,19 +954,13 @@ STATIC ssize_t psmx3_cq_readerr(struct fid_cq *cq, struct fi_cq_err_entry *buf,
...
 	cq_priv->domain->cq_lock_fn(&cq_priv->lock, 2);
 	if (cq_priv->pending_error) {
-		api_version = cq_priv->domain->fabric->util_fabric.
-			      fabric_fid.api_version;
-		size = FI_VERSION_GE(api_version, FI_VERSION(1, 5)) ?
-			sizeof(*buf) : sizeof(struct fi_cq_err_entry_1_0);
-
-		memcpy(buf, &cq_priv->pending_error->cqe, size);
+		ofi_cq_err_memcpy(cq_priv->domain->fabric->util_fabric.fabric_fid.api_version,
+				  buf, &cq_priv->pending_error->cqe);
```
Comment (shefty) [inline prov/psm3/src/psmx3_cq.c:963]: @acgoldma - please verify

File: prov/opx/include/rdma/opx/fi_opx_cq_ops_table.h [hunk 2]
```diff
@@ -72,7 +72,8 @@ fi_opx_cq_readerr(struct fid_cq *cq, struct fi_cq_err_entry *buf, uint64_t flags
 		const int lock_required = fi_opx_threading_lock_required(threading, fi_opx_global.progress);
 
 		fi_opx_lock_if_required(&opx_cq->lock, lock_required);
-		*buf = ext->err_entry;
+		ofi_cq_err_memcpy(opx_cq->domain->fabric->fabric_fid.api_version,
+				  buf, &ext->err_entry);
```
Comment (shefty) [inline prov/opx/include/rdma/opx/fi_opx_cq_ops_table.h:76]: @bcernohous - can you verify the change here and to psm2 are okay

**Comment by iziemba** [comment]
Are the Intel CI failures real?

**Review by bcernohous — APPROVED** [review]

**Comment by shefty** [comment]
The Intel failures do not look related.  Not sure how they would be.

**Pull request merged**